### ISSUE TITLE
Fix to warn that structures not following the Required Context Role

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#816]
- Markuplint version: `3.7.1`
- Parser lang: HTML or JSX and more...
- Node.js version: `18.x`

## Related

- #272
- #673

## Code Example

```html
<div role="table">
  <div role="rowgroup">
    <div role="row">
      <div role="gridcell"></div>
    </div>
  </div>
</div>
```

## The current behavior

No warning

## The expected behavior

`gridcell` is incorrect, `cell` is correct.

**Comment by YusukeHirao** [comment]
## Investigation Report

I investigated this issue by writing verification tests against both ARIA 1.2 and 1.3. See PR #3233 for regression tests.

### HTML element case: Detected

`<table><tbody><tr><td role="gridcell">` is correctly flagged:

> Cannot overwrite the "gridcell" role to the "td" element according to ARIA in HTML specification

This is caught by the **permittedAriaRoles** check. The HTML-ARIA spec defines conditional `permittedRoles` for `<td>`: when inside `<table>` (without `role=grid`/`role=treegrid`), `permittedRoles` is `false`, so any explicit role assignment (including `gridcell`) is rejected. This works in both ARIA 1.2 and 1.3.

### Pure ARIA div case: Not detected

`<div role="table"><div role="row"><div role="gridcell">` produces **no violation**.

This is a structural limitation of the ARIA formal data model, not a markuplint implementation bug:

- `gridcell.requiredContextRole` = `["row"]` — satisfied (parent is `row`)
- `row.allowedAccessibilityChildRoles` includes `gridcell` — satisfied
- **No formal mechanism** distinguishes a `row` inside `table` from a `row` inside `grid`/`treegrid`

The ARIA spec prose says gridcell is *"A cell in a grid or treegrid"*, but this semantic constraint is not encoded in the formal properties (`requiredContextRole`, `requiredOwnedElements`). The formal model treats all `row` contexts identically.

### Conclusion

The reporter's observation is spec-correct: `gridcell` is semantically limited to `grid`/`treegrid` contexts. For native HTML elements, this is already enforced via the HTML-ARIA permittedRoles mechanism. For pure ARIA role assignments on generic elements, detecting this would require custom logic beyond what the ARIA formal data provides — essentially checking the grandparent role chain (`gridcell` → `row` → must be `grid` or `treegrid`, not `table`).

Regression tests added in PR #3233 covering both cases.

**Comment by YusukeHirao** [comment]
Closing: HTML element case is detected via permittedAriaRoles. Pure ARIA div case is a structural limitation of the formal data model. See investigation report above.
